### PR TITLE
Pull request to make the cukes work (especially in SQLite)

### PR DESCRIPTION
**Pull request by MarkDBlackwell**
All tests pass (in SQLite, with Selenium and Webkit). (Also, they pass when set up for PostgreSQL—just for myself.)

Added optional environment variable CAPYBARA_JAVASCRIPT_DRIVER (defaulting to 'selenium') with meaningful settings 'selenium', 'webkit', 'webkit-debug' and 'poltergeist'. (I didn't try Poltergeist.)

Also added optional environment variable FIREFOX_BINARY_PATH.

@tmock12, does this work for you? or would you suggest any changes?

### CODE REVIEW

**Comment by tmock12** [comment]
The Gemfile is maintained by the RailsAppsComposer, not this repo. The ":group => :test" flag was added to the selenium-webdriver gem in https://github.com/RailsApps/rails_apps_composer/commit/a70f5be3fd58c024042be26736b16e52222771f2    The repo just needs to be rebuilt using the composer. 

I like your idea of having an optional javascript driver as I usually use poltergeist myself. But this isn't the best way to do it as it would be useful across all of the apps using the RailsAppsComposer. When you set up these apps using the composer it should be one of the many other options that it lets you select from. This would also get rid of having the other three unused gems in your Gemfile.

**Comment by MarkDBlackwell** [comment]
I agree an optional JavaScript driver would be useful in most apps so generated.

Even a JavaScript driver that is optional at runtime would be useful. This would facilitate selecting Selenium (Firefox) (if desired) while debugging in order to see what is happening in the browser. And otherwise we would use a preference-selectable default JavaScript driver (capybara-webkit or poltergeist) for speed during the usual regression tests.

> Gemfile is maintained by the RailsAppsComposer

@tmock12, so I am aware. :) Your points are quite good regarding integrating with RailsComposer methodology. After utilizing a Rails App for a temporary development platform, generally (I am aware) any changes need to be copied to gem rails_apps_composer.

tl;dr

> The repo just needs to be rebuilt using the composer.

Actually, earlier, I did rebuild this RailsApp from the same commit of gem rails_apps_composer you suggested (at [RailsApps/rails_apps_composer@a70f5be](https://github.com/RailsApps/rails_apps_composer/commit/a70f5be3fd58c024042be26736b16e52222771f2)). Then unfortunately I got an error message, which I now understand was coming from a problem in my Qt installation.

So I switched to a clone of the app from the repository per the [tutorial](https://github.com/RailsApps/rails-prelaunch-signup/blob/master/README.textile#clone). Consistently thereafter I received multiple SQLite database errors about blocking and timeouts, thus prepared this pull request.

Just now as I switch back I observe my regenerated (rails-prelaunch-signup) app is working fine, so I withdraw the pull request. :)
